<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>HypeR-GEM Workflow Illustration • hypeR.GEM</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="HypeR-GEM Workflow Illustration">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hypeR.GEM</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="../articles/Workflow.html">Workflow</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../news/index.html">Change Log</a>
</li>
<li>
  <a href="https://github.com/montilab/hypeR-GEM" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><script src="Workflow_files/core-js-2.5.3/shim.min.js"></script><script src="Workflow_files/react-18.2.0/react.min.js"></script><script src="Workflow_files/react-18.2.0/react-dom.min.js"></script><script src="Workflow_files/reactwidget-2.0.0/react-tools.js"></script><link href="Workflow_files/htmltools-fill-0.5.9/fill.css" rel="stylesheet">
<script src="Workflow_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="Workflow_files/reactable-0.4.5/reactable.css" rel="stylesheet">
<script src="Workflow_files/reactable-binding-0.4.5/reactable.js"></script><script src="Workflow_files/d3-4.9.1/d3.min.js"></script><script src="Workflow_files/sankey-1/sankey.js"></script><script src="Workflow_files/sankeyNetwork-binding-0.4.1/sankeyNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>HypeR-GEM Workflow Illustration</h1>
                        <h4 data-toc-skip class="author">Ziwei
Huang</h4>
            
            <h4 data-toc-skip class="date">12/12/2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/montilab/hypeR-GEM/blob/main/vignettes/Workflow.Rmd" class="external-link"><code>vignettes/Workflow.Rmd</code></a></small>
      <div class="hidden name"><code>Workflow.Rmd</code></div>

    </div>

    
    
<p><font color="#000000"></font></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/load_all.html" class="external-link">load_all</a></span><span class="op">(</span><span class="st">"."</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Loading <span style="color: #0000BB;">hypeR.GEM</span></span></span></code></pre>
<div class="section level2">
<h2 id="what-are-genome-scale-metabolic-modelsgems">What are genome-scale metabolic models(GEMs)?<a class="anchor" aria-label="anchor" href="#what-are-genome-scale-metabolic-modelsgems"></a>
</h2>
<p>Genome-scale metabolic models(GEMs) are curated metabolic models
collect all known metabolic information of a biological
system(e.g. bacteria, human cell, etc.). There are three essential
components in a GEM:</p>
<ul>
<li>
<strong>metabolites</strong>: all known metabolites of a biological
system.</li>
<li>
<strong>reactions</strong>: all known reactions of a biological
system.</li>
<li>
<strong>enzyme-coding genes</strong>: enzymes that catalyze the
biological reactions.</li>
</ul>
<p>Within a GEM, each metabolite is associated with one or more
reactions, in which it participates as either a reactant or a product.
Each reaction, in turn, is catalyzed by one or more enzyme-coding genes,
with exceptions such as exchange reactions and certain transport
reactions that are not catalyzed by enzymes. <strong>HypeR-GEM</strong>
leverages this metabolite–reaction–gene triad to establish a framework
for mapping metabolites to genes.</p>
<p>For more details, see <a href="https://www.pnas.org/doi/abs/10.1073/pnas.2102344118" class="external-link">here</a>.</p>
<p>A schematic representation of a GEM is shown below:</p>
<p><img src="figs/intro_GEM.png" class="r-plt" width="80%" height="80%"></p>
<p>In the diagram, capital letters denote metabolites, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>r</mi><mi>i</mi></msub><annotation encoding="application/x-tex">r_{i}</annotation></semantics></math>
denotes the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>i</mi><mrow><mi>t</mi><mi>h</mi></mrow></msup><annotation encoding="application/x-tex">i^{th}</annotation></semantics></math>
reaction. A prototypical metabolic reaction consists of three
components:</p>
<ul>
<li>
<strong>reactants(substrates)</strong>: one or more metabolites
consumed by the reaction (left-hand side).</li>
<li>
<strong>products</strong>: one or more metabolites produced by the
reaction (right-hand side).</li>
<li>
<strong>catalyst</strong>: enzyme-coding genes whose protein
products catalyze the reaction.</li>
</ul>
<p>For example, reaction
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>r</mi><mn>1</mn></msub><annotation encoding="application/x-tex">r_{1}</annotation></semantics></math>
consumes metabolites
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>A</mi><annotation encoding="application/x-tex">A</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>D</mi><annotation encoding="application/x-tex">D</annotation></semantics></math>
and produces metabolite
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>B</mi><annotation encoding="application/x-tex">B</annotation></semantics></math>,
and it may be catalyzed by one or more enzyme-coding genes.</p>
</div>
<div class="section level2">
<h2 id="mapping-workflow">Mapping workflow<a class="anchor" aria-label="anchor" href="#mapping-workflow"></a>
</h2>
<p>To map a metabolite to its associated enzyme-coding genes,
<strong>hypeR-GEM</strong> defines three types of associations:</p>
<ul>
<li>
<p><strong>metabolite-reaction association</strong>: A metabolite is
associated with a reaction if:</p>
<ul>
<li>The metabolite is either the reactant or the product of the reaction
(<strong>undirectional</strong>)</li>
<li>The metabolite is is the product of the reaction
(<strong>directional</strong>)</li>
</ul>
</li>
<li><p><strong>reaction-enzyme(-coding) gene association</strong>: A
reaction is associated with the enzyme-coding genes that encode enzymes
catalyzing that reaction,</p></li>
<li><p><strong>metabolite-enzyme(-coding) gene association</strong>: A
metabolite is associated with all enzyme-coding genes that catalyze the
reactions in which the metabolite participates.</p></li>
</ul>
<p>The overall workflow is illustrated in the following diagram:
<embed src="figs/mapping.pdf" width="80%" height="80%" type="application/pdf"></embed></p>
<p>Given an input metabolic signature, a set of metabolites annotated
using the <strong>RefMet</strong> nomenclature,
<strong>hypeR-GEM</strong> identifies all metabolic reactions in which
each metabolite participates as either reactant or product. For
reversible reactions, metabolites are treated as both reactants and
products. For each reaction (excluding exchange reactions and a subset
of transport reactions), the associated enzyme-coding genes are then
mapped to the metabolites according to two configurable rules: a
non-directional mapping that considers all reactions involving the
metabolite, and a directional mapping that restricts associations to
reactions producing the metabolite based on the stoichiometry encoded in
the model.</p>
</div>
<div class="section level2">
<h2 id="accounting-for-gene-specificity">Accounting for Gene Specificity<a class="anchor" aria-label="anchor" href="#accounting-for-gene-specificity"></a>
</h2>
<p>Enzyme-coding genes differ substantially in their reaction
specificity. Non-specific (promiscuous) enzymes can catalyze a broad
range of reactions, giving rise to dense gene–reaction–metabolite
connections and consequently weaker, less specific associations with a
given metabolic signature. To quantify gene-level association
specificity and mitigate the influence of such “promiscuous” genes, we
compute a <strong>gene-specificity score</strong> based on the
hypergeometric distribution.</p>
<embed src="figs/gene_specificity.pdf" width="80%" height="80%" type="application/pdf"></embed><p>For each gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>g</mi><mi>i</mi></msub><annotation encoding="application/x-tex">g_{i}</annotation></semantics></math>,
let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
denote he total number of metabolites represented in the GEM,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>S</mi><annotation encoding="application/x-tex">S</annotation></semantics></math>
the number of metabolites in the metabolic signature,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>A</mi><mi>i</mi></msub><annotation encoding="application/x-tex">A_{i}</annotation></semantics></math>
the number of metabolites associated with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>g</mi><mi>i</mi></msub><annotation encoding="application/x-tex">g_{i}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>a</mi><mi>i</mi></msub><annotation encoding="application/x-tex">a_{i}</annotation></semantics></math>
the number of those metabolites that overlap with the signature. The
significance score
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>s</mi><mi>i</mi></msub><annotation encoding="application/x-tex">s_{i}</annotation></semantics></math>
is defined as the upper-tail hypergeometric probability of observing at
least
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>a</mi><mi>i</mi></msub><annotation encoding="application/x-tex">a_{i}</annotation></semantics></math>
overlapping metabolites by chance:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>s</mi><mi>i</mi></msub><mo>=</mo><mi>P</mi><mo stretchy="false" form="prefix">(</mo><mi>X</mi><mo>≥</mo><msub><mi>a</mi><mi>i</mi></msub><mo stretchy="false" form="postfix">)</mo><mo>=</mo><munderover><mo>∑</mo><mrow><mi>x</mi><mo>=</mo><msub><mi>a</mi><mi>i</mi></msub></mrow><mrow><mrow><mi mathvariant="normal">min</mi><mo>⁡</mo></mrow><mo stretchy="false" form="prefix">(</mo><msub><mi>A</mi><mi>i</mi></msub><mo>,</mo><mi>S</mi><mo stretchy="false" form="postfix">)</mo></mrow></munderover><mfrac><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mfrac linethickness="0"><msub><mi>A</mi><mi>i</mi></msub><mi>x</mi></mfrac><mo stretchy="true" form="postfix">)</mo></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mfrac linethickness="0"><mrow><mi>M</mi><mo>−</mo><msub><mi>A</mi><mi>i</mi></msub></mrow><mrow><mi>S</mi><mo>−</mo><mi>x</mi></mrow></mfrac><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mfrac linethickness="0"><mi>M</mi><mi>S</mi></mfrac><mo stretchy="true" form="postfix">)</mo></mrow></mfrac></mrow><annotation encoding="application/x-tex">
s_i = P(X \ge a_i)
= \sum_{x = a_i}^{\min(A_i, S)}
\frac{\binom{A_i}{x}\binom{M - A_i}{S - x}}{\binom{M}{S}}
</annotation></semantics></math></p>
<p>Smaller values of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>s</mi><mi>i</mi></msub><annotation encoding="application/x-tex">s_{i}</annotation></semantics></math>
a more specific association between gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>g</mi><mi>i</mi></msub><annotation encoding="application/x-tex">g_{i}</annotation></semantics></math>
and the metabolic signature.</p>
</div>
<div class="section level2">
<h2 id="sigmoid-based-weighting-of-gene-specificity">Sigmoid-based weighting of gene specificity<a class="anchor" aria-label="anchor" href="#sigmoid-based-weighting-of-gene-specificity"></a>
</h2>
<p>To incorporate gene-specificity information into downstream
enrichment analyses, each gene is assigned a continuous weight
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>w</mi><mi>i</mi></msub><mo>∈</mo><mo stretchy="false" form="prefix">[</mo><mn>0</mn><mo>,</mo><mn>1</mn><mo stretchy="false" form="postfix">]</mo></mrow><annotation encoding="application/x-tex">w_{i} \in [0, 1]</annotation></semantics></math>
via a sigmoid transformation of its significance score:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>w</mi><mi>i</mi></msub><mo>=</mo><mfrac><mn>1</mn><mrow><mn>1</mn><mo>+</mo><msup><mi>e</mi><mrow><mi>a</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mrow><mi mathvariant="normal">log</mi><mo>⁡</mo></mrow><mn>10</mn></msub><mo stretchy="false" form="prefix">(</mo><msub><mi>s</mi><mi>i</mi></msub><mo stretchy="false" form="postfix">)</mo><mo>−</mo><mi>b</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow></msup></mrow></mfrac></mrow><annotation encoding="application/x-tex">
w_i = \frac{1}{1 + e^{a\left(\log_{10}(s_i) - b\right)}}
</annotation></semantics></math> This transformation is
<strong>monotonic and bounded</strong>, ensuring that genes with
stronger specificity (smaller
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>s</mi><mi>i</mi></msub><annotation encoding="application/x-tex">s_{i}</annotation></semantics></math>)
receive larger weights, while genes with weak or nonspecific
associations are smoothly down-weighted.</p>
<p>The parameter
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>b</mi><annotation encoding="application/x-tex">b</annotation></semantics></math>
determines the <strong>inflection point</strong> of the curve (the
half-point, where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>w</mi><mi>i</mi></msub><mo>=</mo><mn>0.5</mn></mrow><annotation encoding="application/x-tex">w_{i}=0.5</annotation></semantics></math>),
with default value
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>b</mi><mo>=</mo><mi>−</mi><mn>1</mn></mrow><annotation encoding="application/x-tex">b=-1</annotation></semantics></math>
allowing users to control the weight threshold. By default,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>b</mi><mo>=</mo><mi>−</mi><mn>1</mn></mrow><annotation encoding="application/x-tex">b=-1</annotation></semantics></math>,
corresponding to a half-point at
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>s</mi><mi>i</mi></msub><mo>=</mo><mn>0.1</mn></mrow><annotation encoding="application/x-tex">s_{i} = 0.1</annotation></semantics></math>.</p>
<p>The parameter
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>a</mi><mo>&gt;</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">a&gt;0</annotation></semantics></math>
regulates the <strong>steepness</strong> of the curve. Larger values of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>a</mi><annotation encoding="application/x-tex">a</annotation></semantics></math>
produce a sharper transition, whereas smaller values yield a smoother
weighting scheme. The default value
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>a</mi><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">a=1</annotation></semantics></math>
provides a balanced compromise between sensitivity and smoothness.</p>
</div>
<div class="section level2">
<h2 id="alternative-weighting-scheme">Alternative weighting scheme<a class="anchor" aria-label="anchor" href="#alternative-weighting-scheme"></a>
</h2>
<p>For additional flexibility, <strong>hypeR-GEM</strong> also supports
a linear weighting scheme based directly on the significance score:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>w</mi><mi>i</mi></msub><mo>=</mo><mn>1</mn><mo>−</mo><msub><mi>s</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">
w_i = 1 - s_{i}
</annotation></semantics></math> which assigns weights proportional to
the complement of the gene-specificity p-value (adjusted p-value).</p>
</div>
<div class="section level2">
<h2 id="weighted-hypergeometric-test">Weighted hypergeometric test<a class="anchor" aria-label="anchor" href="#weighted-hypergeometric-test"></a>
</h2>
<p>The standard hypergeometric test used in over-representation analysis
(ORA) evaluates whether a gene set contains more genes associated with a
given category (e.g., pathway or ontology term) than expected by chance,
implicitly assuming that all genes contribute equally to the enrichment.
However, as described in the <em>Gene Specificity</em> section,
enzyme-coding genes mapped through genome-scale metabolic models (GEMs)
can differ substantially in their association specificity with a given
metabolic signature. To incorporate this heterogeneity, hypeR-GEM
extends the standard hypergeometric test to a weighted formulation that
accounts for gene-specific association strengths.</p>
<p>Formally, the probability of observing at least
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
overlapping genes between a gene set and the input signature under the
standard hypergeometric model is given by:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo stretchy="false" form="prefix">(</mo><mi>X</mi><mo>≥</mo><mi>k</mi><mo stretchy="false" form="postfix">)</mo><mo>=</mo><munderover><mo>∑</mo><mrow><mi>j</mi><mo>=</mo><mi>k</mi></mrow><mi>n</mi></munderover><mfrac><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mfrac linethickness="0"><mi>K</mi><mi>j</mi></mfrac><mo stretchy="true" form="postfix">)</mo></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mfrac linethickness="0"><mrow><mi>N</mi><mo>−</mo><mi>K</mi></mrow><mrow><mi>n</mi><mo>−</mo><mi>j</mi></mrow></mfrac><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mfrac linethickness="0"><mi>N</mi><mi>n</mi></mfrac><mo stretchy="true" form="postfix">)</mo></mrow></mfrac><mo>,</mo></mrow><annotation encoding="application/x-tex">
P(X \ge k)
= \sum_{j = k}^{n}
\frac{\binom{K}{j}\binom{N - K}{n - j}}{\binom{N}{n}},
</annotation></semantics></math></p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>N</mi><annotation encoding="application/x-tex">N</annotation></semantics></math>
denotes the background size,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>n</mi><annotation encoding="application/x-tex">n</annotation></semantics></math>
is the number of GEM-mapped enzyme-coding genes in the signature,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>K</mi><annotation encoding="application/x-tex">K</annotation></semantics></math>
is the size of the gene set, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
is the number of overlapping genes.</p>
<p>In the weighted formulation, each gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
is assigned a weight
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>w</mi><mi>i</mi></msub><mo>∈</mo><mo stretchy="false" form="prefix">[</mo><mn>0</mn><mo>,</mo><mn>1</mn><mo stretchy="false" form="postfix">]</mo></mrow><annotation encoding="application/x-tex">w_i \in [0,1]</annotation></semantics></math>
reflecting its association specificity. The effective signature size and
overlap are then defined as continuous, weight-adjusted counts:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>n</mi><mi>w</mi></msub><mo>=</mo><munder><mo>∑</mo><mrow><mi>i</mi><mo>∈</mo><mi>n</mi></mrow></munder><msub><mi>w</mi><mi>i</mi></msub><mo>,</mo><mspace width="2.0em"></mspace><msub><mi>k</mi><mi>w</mi></msub><mo>=</mo><munder><mo>∑</mo><mrow><mi>i</mi><mo>∈</mo><mi>k</mi></mrow></munder><msub><mi>w</mi><mi>i</mi></msub><mi>.</mi></mrow><annotation encoding="application/x-tex">
n_w = \sum_{i \in n} w_i,
\qquad
k_w = \sum_{i \in k} w_i.
</annotation></semantics></math></p>
<p>Because the hypergeometric distribution is defined for integer-valued
inputs, these quantities are rounded to the nearest integer using
standard rounding:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>n</mi><mi>w</mi></msub><mo>=</mo><mrow><mo stretchy="true" form="prefix">⌊</mo><mstyle displaystyle="false"><mfrac><mn>1</mn><mn>2</mn></mfrac></mstyle><mo>+</mo><munder><mo>∑</mo><mrow><mi>i</mi><mo>∈</mo><mi>n</mi></mrow></munder><msub><mi>w</mi><mi>i</mi></msub><mo stretchy="true" form="postfix">⌋</mo></mrow><mo>,</mo><mspace width="2.0em"></mspace><msub><mi>k</mi><mi>w</mi></msub><mo>=</mo><mrow><mo stretchy="true" form="prefix">⌊</mo><mstyle displaystyle="false"><mfrac><mn>1</mn><mn>2</mn></mfrac></mstyle><mo>+</mo><munder><mo>∑</mo><mrow><mi>i</mi><mo>∈</mo><mi>k</mi></mrow></munder><msub><mi>w</mi><mi>i</mi></msub><mo stretchy="true" form="postfix">⌋</mo></mrow><mo>,</mo></mrow><annotation encoding="application/x-tex">
n_w = \left\lfloor \tfrac{1}{2} + \sum_{i \in n} w_i \right\rfloor,
\qquad
k_w = \left\lfloor \tfrac{1}{2} + \sum_{i \in k} w_i \right\rfloor,
</annotation></semantics></math></p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="false" form="prefix">⌊</mo><mi>⋅</mi><mo stretchy="false" form="postfix">⌋</mo></mrow><annotation encoding="application/x-tex">\lfloor \cdot \rfloor</annotation></semantics></math>
denotes the floor operator.</p>
<p>Substituting
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>n</mi><mi>w</mi></msub><annotation encoding="application/x-tex">n_w</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>k</mi><mi>w</mi></msub><annotation encoding="application/x-tex">k_w</annotation></semantics></math>
into the hypergeometric test yields a weighted enrichment score that
down-weights highly promiscuous genes with low specificity, thereby
reducing noise and limiting spurious enrichments. This weighting
strategy improves the robustness, biological relevance, and
interpretability of enrichment results derived from GEM-based
metabolite-to-gene mappings.</p>
</div>
<div class="section level2">
<h2 id="workflow-illustration">Workflow Illustration<a class="anchor" aria-label="anchor" href="#workflow-illustration"></a>
</h2>
<p>Here, we illustrate the workflow using metabolite signatures
associated with coronavirus disease 2019 (COVID-19), derived from human
urine samples(See <a href="https://www.sciencedirect.com/science/article/pii/S2211124721017836#da0010" class="external-link">reference</a>).
This study profiled the urinary metabolome of 71 patients with COVID-19
and 27 healthy controls. The publicly available COVID-19–associated
metabolites were classified into two signatures: up-regulated and
down-regulated.</p>
<div class="section level3">
<h3 id="load-metabolite-signature">Load metabolite signature<a class="anchor" aria-label="anchor" href="#load-metabolite-signature"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">COVID_urine</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="hyper-gem-mapping">HypeR-GEM mapping<a class="anchor" aria-label="anchor" href="#hyper-gem-mapping"></a>
</h3>
<p>The core function in <strong>hypeR-GEM</strong> is
<code><a href="../reference/signature2gene.html">signature2gene()</a></code>, which maps metabolite signatures to
enzyme-coding genes using GEMs. The function takes the following key
arguments:</p>
<ul>
<li><p><code>signature</code>: A list of metabolic signatures. Each
element must be a data frame containing a column whose name matches the
<code>reference_key</code> argument, , which specifies the
<strong>RefMet</strong> annotation for each metabolite.</p></li>
<li>
<p><code>directional</code>: Logical argument specifying the
metabolite–reaction mapping rule.</p>
<ul>
<li>If <code>TRUE</code>, metabolites are mapped only to reactions in
which they appear as products (directional mapping).</li>
<li>If <code>FALSE</code>, metabolites are mapped to reactions in which
they appear as either reactants or products (non-directional
mapping).</li>
</ul>
</li>
<li><p><code>promiscuous_threshold</code>: Maximum allowable number of
metabolites associated with a gene. Genes exceeding this threshold are
classified as promiscuous and excluded from downstream
analysis.</p></li>
<li><p><code>background</code>: Background used in the gene-specific
hypergeometric test. By default, this is set to the total number of
metabolites represented in the GEM.</p></li>
<li><p><code>reference_key</code>: A character string indicating the
column name in each signature data frame that contains
<strong>RefMet</strong> metabolite identifiers.</p></li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Undirectional mapping</span></span>
<span><span class="va">hypeR_GEM_obj</span> <span class="op">&lt;-</span> <span class="fu">hypeR.GEM</span><span class="fu">::</span><span class="fu"><a href="../reference/signature2gene.html">signature2gene</a></span><span class="op">(</span>signatures <span class="op">=</span> <span class="va">COVID_urine</span>, </span>
<span>                                           directional <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                           promiscuous_threshold <span class="op">=</span> <span class="fl">100</span>, </span>
<span>                                           reference_key <span class="op">=</span> <span class="st">'refmet_name'</span>,</span>
<span>                                           background <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="co">##Directional mapping</span></span>
<span><span class="va">hypeR_GEM_obj_di</span> <span class="op">&lt;-</span> <span class="fu">hypeR.GEM</span><span class="fu">::</span><span class="fu"><a href="../reference/signature2gene.html">signature2gene</a></span><span class="op">(</span>signatures <span class="op">=</span> <span class="va">COVID_urine</span>,</span>
<span>                                              directional <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                              promiscuous_threshold <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                                              reference_key <span class="op">=</span> <span class="st">'refmet_name'</span>,</span>
<span>                                              background <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 class="tabset" id="hyper-gem-output">HypeR-GEM output<a class="anchor" aria-label="anchor" href="#hyper-gem-output"></a>
</h3>
<p>Here, we illustrate the output of <strong>hypeR-GEM</strong> using
the results from non-directional mapping. The output of
<code><a href="../reference/signature2gene.html">hypeR.GEM::signature2gene()</a></code> is a structured list comprising
three major components:</p>
<ul>
<li><p><code>promiscuous_met</code>: A data frame listing metabolites
considered promiscuous, i.e., those associated with an excessively large
number of genes. The filtering threshold is defined by the
<code>promiscuous_threshold</code> parameter in
<code><a href="../reference/signature2gene.html">signature2gene()</a></code>.</p></li>
<li><p><code>mapped_metabolite_signatures</code>: A list in which each
element corresponds to a metabolite signature. Each element is a data
frame that provides the identifier of the metabolite in the GEM
(<code>name</code>), its full name (<code>fullname</code>), the
corresponding RefMet annotation (<code>refmet_name</code>), and the
number of genes associated with that metabolite
(<code>gene_association</code>).</p></li>
<li>
<p><code>gene_table</code>: The core output of the mapping
procedure. This is a list where each element corresponds to a metabolite
signature and contains a data frame of the mapped enzyme-coding genes.
For each gene, the data frame reports:</p>
<ul>
<li>the number of catalyzed reactions
(<code>associated_reactions</code>)</li>
<li>the total number of associated metabolites in the GEM
(<code>total_association</code>)</li>
<li>the number of associated metabolites within the given signature
(<code>signature_association</code>)</li>
<li>the p-value quantifying the mapping specificity
(<code>p_value</code>)</li>
</ul>
</li>
</ul>
<div class="section level4">
<h4 id="promiscuous_met">promiscuous_met<a class="anchor" aria-label="anchor" href="#promiscuous_met"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hypeR_GEM_obj</span><span class="op">$</span><span class="va">promiscuous_met</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            refmet_name gene_association</span></span>
<span><span class="co">## 1        Glutamic acid              109</span></span>
<span><span class="co">## 2 S-Adenosylmethionine              152</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="mapped_metabolite_signatures">mapped_metabolite_signatures<a class="anchor" aria-label="anchor" href="#mapped_metabolite_signatures"></a>
</h4>
<p>Given that we have two metabolite signatures, namely up-regulated and
down-regulated, the <code>mapped_metabolite_signatures</code> object is
a list of length two. For illustration, we focus here on the
up-regulated signature as an example.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hypeR_GEM_obj</span><span class="op">$</span><span class="va">mapped_metabolite_signatures</span><span class="op">$</span><span class="va">up</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       name               fullname                refmet_name gene_association</span></span>
<span><span class="co">## 1 MAM00654 2-hydroxyphenylacetate 2-Hydroxyphenylacetic acid                2</span></span>
<span><span class="co">## 2 MAM00664 2-methylbutyrylglycine     2-Methylbutyrylglycine                4</span></span>
<span><span class="co">## 3 MAM00672       2-oxoglutaramate     2-Keto-glutaramic acid                4</span></span>
<span><span class="co">## 4 MAM00775  3-hydroxyanthranilate  3-Hydroxyanthranilic acid                3</span></span>
<span><span class="co">## 5 MAM00788 3-hydroxy-L-kynurenine          Hydroxykynurenine                7</span></span>
<span><span class="co">## 6 MAM00952   4-acetamidobutanoate   4-Acetamidobutanoic acid                6</span></span>
<span><span class="co">##      formula exact_mass   super_class               main_class</span></span>
<span><span class="co">## 1     C8H8O3   152.0473 Organic acids         Phenylpropanoids</span></span>
<span><span class="co">## 2   C7H13NO3   159.0895 Organic acids Amino acids and peptides</span></span>
<span><span class="co">## 3    C5H7NO4   145.0375 Organic acids               Keto acids</span></span>
<span><span class="co">## 4    C7H7NO3   153.0426    Benzenoids                 Benzenes</span></span>
<span><span class="co">## 5 C10H12N2O4   224.0797 Organic acids Amino acids and peptides</span></span>
<span><span class="co">## 6   C6H11NO3   145.0739 Organic acids Amino acids and peptides</span></span>
<span><span class="co">##                sub_class</span></span>
<span><span class="co">## 1         Cinnamic acids</span></span>
<span><span class="co">## 2            Amino acids</span></span>
<span><span class="co">## 3 Short-chain keto acids</span></span>
<span><span class="co">## 4   Hydroxybenzoic acids</span></span>
<span><span class="co">## 5            Amino acids</span></span>
<span><span class="co">## 6            Amino acids</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="gene_table">gene_table<a class="anchor" aria-label="anchor" href="#gene_table"></a>
</h4>
<p>Similarly, the <code>gene_table</code> is also a list object of
length two. For illustration, we focus here on the mapped enzyme-coding
genes derived from the up-regulated metabolite signature as an
example.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hypeR_GEM_obj</span><span class="op">$</span><span class="va">gene_tables</span><span class="op">$</span><span class="va">up</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">##              name  symbol associated_reactions total_association</span></span>
<span><span class="co">## 1 ENSG00000158104     HPD                    2                 6</span></span>
<span><span class="co">## 2 ENSG00000155380 SLC16A1                   80                20</span></span>
<span><span class="co">## 3 ENSG00000149124   GLYAT                   11                23</span></span>
<span><span class="co">## 4 ENSG00000156689 GLYATL2                    5                13</span></span>
<span><span class="co">## 5 ENSG00000166840 GLYATL1                    5                13</span></span>
<span><span class="co">## 6 ENSG00000165970  SLC6A5                    5                 6</span></span>
<span><span class="co">##   signature_association signature_size background      p_value         fdr</span></span>
<span><span class="co">## 1                     1             49       4112 0.0020241802 0.003844373</span></span>
<span><span class="co">## 2                     1             49       4112 0.0230551953 0.025443797</span></span>
<span><span class="co">## 3                     3             49       4112 0.0001334704 0.001090009</span></span>
<span><span class="co">## 4                     2             49       4112 0.0004183142 0.002440166</span></span>
<span><span class="co">## 5                     2             49       4112 0.0004183142 0.002440166</span></span>
<span><span class="co">## 6                     1             49       4112 0.0020241802 0.003844373</span></span>
<span><span class="co">##   one_minus_p_value one_minus_fdr</span></span>
<span><span class="co">## 1         0.9979758     0.9961556</span></span>
<span><span class="co">## 2         0.9769448     0.9745562</span></span>
<span><span class="co">## 3         0.9998665     0.9989100</span></span>
<span><span class="co">## 4         0.9995817     0.9975598</span></span>
<span><span class="co">## 5         0.9995817     0.9975598</span></span>
<span><span class="co">## 6         0.9979758     0.9961556</span></span>
<span><span class="co">##                                    associated_metabolites</span></span>
<span><span class="co">## 1                              2-Hydroxyphenylacetic acid</span></span>
<span><span class="co">## 2                              2-Hydroxyphenylacetic acid</span></span>
<span><span class="co">## 3 2-Methylbutyrylglycine;Isobutyrylglycine;Suberylglycine</span></span>
<span><span class="co">## 4                2-Methylbutyrylglycine;Isobutyrylglycine</span></span>
<span><span class="co">## 5                2-Methylbutyrylglycine;Isobutyrylglycine</span></span>
<span><span class="co">## 6                                  2-Methylbutyrylglycine</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 class="tabset" id="enrichment-analysis">Enrichment analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h3>
<p>Here, we use the REACTOME gene sets as an example to illustrate the
downstream enrichment analysis of enzyme-coding genes mapped by
hypeR-GEM.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">reactome</span><span class="op">)</span></span></code></pre></div>
<p>In this example, the background is defined as all enzyme-coding genes
represented in the Human GEM model (see <a href="https://www.pnas.org/doi/abs/10.1073/pnas.2102344118" class="external-link">reference</a>).
The <code><a href="../reference/enrichment.html">enrichment()</a></code> function performs gene-set enrichment
analysis and accepts the following key arguments:</p>
<ul>
<li><p><code>hypeR_GEM_obj</code>: The object returned by
<code><a href="../reference/signature2gene.html">signature2gene()</a></code> function.</p></li>
<li><p><code>genesets</code>: A list of gene-sets</p></li>
<li><p><code>genesets_name</code>: A descriptive name for the gene-set
compendium (e.g., “REACTOME”).</p></li>
<li><p><code>method</code>: Enrichment method. “unweighted” applies the
standard Fisher/hypergeometric test, while “weighted” applies the
weighted hypergeometric test.</p></li>
<li><p><code>weighted_by</code>: Used only when
<code>method = "weighted"</code>. Specifies the column in
<code>hyper_GEM_obj[["gene_table"]]</code> containing the gene-specific
significance score
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>s</mi><mi>i</mi></msub><annotation encoding="application/x-tex">s_{i}</annotation></semantics></math>.</p></li>
<li><p><code>sigmoid_transformation</code>: Logical. If
<code>TRUE</code>, applies the sigmoid transformation to
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>s</mi><mi>i</mi></msub><annotation encoding="application/x-tex">s_{i}</annotation></semantics></math>,
if <code>FALSE</code>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>1</mn><mo>−</mo><msub><mi>s</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">1-s_{i}</annotation></semantics></math>
is used.</p></li>
<li><p><code>min_metabolite</code>: Additional filtering criterion that
specifies the minimum number of metabolites required for enrichment for
account for metabolite specificity.</p></li>
</ul>
<p>Downstream visualization is supported by the
<code><a href="../reference/enrichment_plot.html">enrichment_plot()</a></code> and <code><a href="../reference/rctbls.html">rctbls()</a></code> functions.</p>
<p>In our example, results obtained using the non-directional mapping
rule show that pathways such as <em>Sialic acid metabolism</em>,
<em>Trafficking and processing of endosomal TLR</em>, and <em>Choline
catabolism</em> are enriched only under the unweighted test, suggesting
that these signals are primarily driven by non-specific genes.</p>
<div class="section level4">
<h4 id="unweighted-standard-hypergeometric-test">Unweighted (standard) hypergeometric test<a class="anchor" aria-label="anchor" href="#unweighted-standard-hypergeometric-test"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">max_fdr</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co">## Enrichment analysis from undirectional mapping</span></span>
<span><span class="va">enrichment_obj</span> <span class="op">&lt;-</span> <span class="fu">hypeR.GEM</span><span class="fu">::</span><span class="fu"><a href="../reference/enrichment.html">enrichment</a></span><span class="op">(</span><span class="va">hypeR_GEM_obj</span>,</span>
<span>                                        genesets <span class="op">=</span> <span class="va">reactome</span>,</span>
<span>                                        genesets_name <span class="op">=</span> <span class="st">"REACTOME"</span>,</span>
<span>                                        method<span class="op">=</span><span class="st">'unweighted'</span>,</span>
<span>                                        min_metabolite <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                        background<span class="op">=</span><span class="fl">3068</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">hypeR.GEM</span><span class="fu">::</span><span class="fu"><a href="../reference/enrichment_plot.html">enrichment_plot</a></span><span class="op">(</span><span class="va">enrichment_obj</span>,</span>
<span>                                top<span class="op">=</span><span class="fl">40</span>,</span>
<span>                                abrv<span class="op">=</span><span class="fl">50</span>,</span>
<span>                                size_by<span class="op">=</span><span class="st">"significance"</span>,</span>
<span>                                fdr_cutoff<span class="op">=</span> <span class="va">max_fdr</span>,</span>
<span>                                val<span class="op">=</span><span class="st">'fdr'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"FDR ≤"</span>, <span class="va">max_fdr</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<p><img src="Workflow_files/figure-html/unnamed-chunk-11-1.png" class="r-plt" width="700"></p>
</div>
<div class="section level4">
<h4 id="reactable-table">Reactable table<a class="anchor" aria-label="anchor" href="#reactable-table"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/rctbls.html">rctbls</a></span><span class="op">(</span><span class="va">enrichment_obj</span><span class="op">)</span></span></code></pre></div>
<div class="rctbl-outer-obj">
<div class="reactable html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"signature":["up","dn"],"size":[245,200],"enriched":[10,3],"gsets":["REACTOME","REACTOME"],"bg":[3068,3068]},"columns":[{"id":".details","name":"","type":null,"headerClassName":"rctbl-outer-header","sortable":false,"resizable":false,"filterable":false,"searchable":false,"width":45,"align":"center","details":[{"name":"div","attribs":{"className":"rctbl-inner-obj"},"children":[{"name":"Reactable","attribs":{"data":{"label":["Amino acid transport across the plasma membrane","Tryptophan catabolism","Aspartate and asparagine metabolism","Recycling of bile acids and salts","Urea cycle","Phenylalanine metabolism","Pyrimidine catabolism","Sialic acid metabolism","Serotonin and melatonin biosynthesis","Trafficking and processing of endosomal TLR"],"pval":[2.1e-23,6.7e-09,5.5e-07,8e-07,1.9e-06,1.8e-05,0.00015,0.00016,0.00019,0.00026],"fdr":[3.3e-20,5.2e-06,0.00029,0.00031,0.00058,0.0046,0.031,0.031,0.032,0.041],"signature":[245,245,245,245,245,245,245,245,245,245],"geneset":[33,14,12,16,10,6,12,33,5,13],"overlap":[26,10,8,9,7,5,6,10,4,6],"gene_hits":["SLC16A10;SLC1A4;SLC36A1;SLC36A2;SLC36A4;SLC38A1;SLC38A2;SLC38A4;SLC3A1;SLC3A2;SLC43A1;SLC43A2;SLC6A14;SLC6A15;SLC6A18;SLC6A19;SLC6A20;SLC7A1;SLC7A11;SLC7A2;SLC7A3;SLC7A5;SLC7A6;SLC7A7;SLC7A8;SLC7A9","AADAT;AFMID;HAAO;KMO;KYAT1;KYAT3;KYNU;SLC36A4;SLC3A2;SLC7A5","ASNS;ASPA;ASPG;GOT1;GOT2;NAT8L;SLC25A12;SLC25A13","ABCB11;ABCC3;ALB;BAAT;SLC10A1;SLC10A2;SLCO1A2;SLCO1B1;SLCO1B3","ARG1;ARG2;ASS1;NAGS;OTC;SLC25A15;SLC25A2","ASRGL1;KYAT1;PAH;PCBD1;QDPR","NT5C;NT5C1A;NT5C3A;NT5E;NT5M;TYMP","CMAS;CTSA;GLB1;NANP;NEU1;NEU2;NEU3;NEU4;NPL;SLC17A5","ASMT;DDC;TPH1;TPH2","CTSB;CTSK;CTSL;CTSS;CTSV;LGMN"],"metabolite_hits":["Hydroxykynurenine;Ornithine;Phenylalanine;Proline;Aspartic acid;Cystine;Glycocholic acid;Kynurenine","2-Keto-glutaramic acid;Hydroxykynurenine;Kynurenic acid;Kynurenine;3-Hydroxyanthranilic acid;Ornithine;Phenylalanine;Proline;Cystine","Aspartic acid;Phenylalanine","Aspartic acid;Phenylalanine;Proline;Glycochenodeoxycholic acid;Glycocholic acid;Glycoursodeoxycholic acid;Glycochenodeoxycholic acid 3-sulfate;Glycodeoxycholic acid 3-sulfate;Taurochenodeoxycholic acid 3-sulfate;Taurodeoxycholic acid 3-sulfate","Aspartic acid;N-Acetylmethionine;Ornithine","2-Keto-glutaramic acid;Hydroxykynurenine;Kynurenic acid;Kynurenine;Aspartic acid;4a-Carbinolamine tetrahydrobiopterin;Phenylalanine","Deoxyguanosine;Guanosine;Deoxyinosine","Aspartic acid;N-Acetylneuraminic acid;Phenylalanine;Proline;Glucuronic acid","Hydroxykynurenine;Phenylalanine;5-Hydroxyindoleacetic acid;4a-Carbinolamine tetrahydrobiopterin","Aspartic acid;Phenylalanine;Proline"]},"columns":[{"id":"label","name":"Label","type":"character","headerClassName":"rctbl-inner-header","minWidth":300},{"id":"pval","name":"P-Value","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"fdr","name":"FDR","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"signature","name":"signature","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"geneset","name":"Geneset Size","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"overlap","name":"Overlap","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"gene_hits","name":"Gene Hits","type":"character","headerClassName":"rctbl-inner-header"},{"id":"metabolite_hits","name":"metabolite_hits","type":"character","headerClassName":"rctbl-inner-header"}],"resizable":true,"searchable":true,"showPageSizeOptions":false,"compact":true,"className":"rctbl-inner-tbl","dataKey":"f50663211317675a7905fe5bdf4c0851","static":false,"nested":true},"children":[]}]},{"name":"div","attribs":{"className":"rctbl-inner-obj"},"children":[{"name":"Reactable","attribs":{"data":{"label":["Pyrimidine catabolism","Transport of nucleosides and free purine and pyrimidine bases across the plasma membrane","Choline catabolism"],"pval":[2.7e-06,2.7e-06,0.00024],"fdr":[0.00071,0.00071,0.046],"signature":[200,200,200],"geneset":[12,12,6],"overlap":[7,7,4],"gene_hits":["DPYD;NT5C;NT5C1A;NT5C3A;NT5E;NT5M;TYMP","SLC28A1;SLC28A2;SLC28A3;SLC29A1;SLC29A2;SLC29A3;SLC29A4","ALDH7A1;BHMT;DMGDH;SARDH"],"metabolite_hits":["Adenosine;D-Ribosylnicotinate;Thymine","Adenosine;Serotonin;Tryptamine;Tyramine;Thymine","Dimethylglycine;Methylmalonic acid"]},"columns":[{"id":"label","name":"Label","type":"character","headerClassName":"rctbl-inner-header","minWidth":300},{"id":"pval","name":"P-Value","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"fdr","name":"FDR","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"signature","name":"signature","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"geneset","name":"Geneset Size","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"overlap","name":"Overlap","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"gene_hits","name":"Gene Hits","type":"character","headerClassName":"rctbl-inner-header"},{"id":"metabolite_hits","name":"metabolite_hits","type":"character","headerClassName":"rctbl-inner-header"}],"resizable":true,"searchable":true,"showPageSizeOptions":false,"compact":true,"className":"rctbl-inner-tbl","dataKey":"0657b90e433bf9cb427e3998105cee86","static":false,"nested":true},"children":[]}]}]},{"id":"signature","name":"Signature","type":"character","headerClassName":"rctbl-outer-header","minWidth":300},{"id":"size","name":"Signature Size","type":"numeric","headerClassName":"rctbl-outer-header"},{"id":"enriched","name":"Enriched Genesets","type":"numeric","headerClassName":"rctbl-outer-header"},{"id":"gsets","name":"Genesets","type":"character","headerClassName":"rctbl-outer-header"},{"id":"bg","name":"Background","type":"numeric","headerClassName":"rctbl-outer-header"}],"resizable":true,"showPageSizeOptions":false,"onClick":"expand","nowrap":true,"className":"rctbl-outer-tbl","rowStyle":{"cursor":"pointer"},"dataKey":"183b0a7c3ed7d945324d25655021d7d2"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="section level4">
<h4 id="weighted-hypergeometric-test-1">Weighted hypergeometric test<a class="anchor" aria-label="anchor" href="#weighted-hypergeometric-test-1"></a>
</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">max_fdr</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co">## Enrichment analysis from undirectional mapping</span></span>
<span><span class="va">enrichment_obj_wt</span> <span class="op">&lt;-</span> <span class="fu">hypeR.GEM</span><span class="fu">::</span><span class="fu"><a href="../reference/enrichment.html">enrichment</a></span><span class="op">(</span><span class="va">hypeR_GEM_obj</span>,</span>
<span>                                          genesets <span class="op">=</span> <span class="va">reactome</span>,</span>
<span>                                          genesets_name <span class="op">=</span> <span class="st">"REACTOME"</span>,</span>
<span>                                          method<span class="op">=</span><span class="st">'weighted'</span>,</span>
<span>                                          weighted_by <span class="op">=</span> <span class="st">'fdr'</span>,</span>
<span>                                          sigmoid_transformation <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                          min_metabolite <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                          background<span class="op">=</span><span class="fl">3068</span><span class="op">)</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">hypeR.GEM</span><span class="fu">::</span><span class="fu"><a href="../reference/enrichment_plot.html">enrichment_plot</a></span><span class="op">(</span><span class="va">enrichment_obj_wt</span>,</span>
<span>                                top<span class="op">=</span><span class="fl">40</span>,</span>
<span>                                abrv<span class="op">=</span><span class="fl">50</span>,</span>
<span>                                size_by<span class="op">=</span><span class="st">"significance"</span>,</span>
<span>                                fdr_cutoff<span class="op">=</span> <span class="va">max_fdr</span>,</span>
<span>                                val<span class="op">=</span><span class="st">'fdr'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"FDR ≤"</span>, <span class="va">max_fdr</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<p><img src="Workflow_files/figure-html/unnamed-chunk-13-1.png" class="r-plt" width="700"></p>
</div>
<div class="section level4">
<h4 id="reactable-table-1">Reactable table<a class="anchor" aria-label="anchor" href="#reactable-table-1"></a>
</h4>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/rctbls.html">rctbls</a></span><span class="op">(</span><span class="va">enrichment_obj_wt</span><span class="op">)</span></span></code></pre></div>
<div class="rctbl-outer-obj">
<div class="reactable html-widget html-fill-item" id="htmlwidget-e5c8c404fe174e4c81bd" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e5c8c404fe174e4c81bd">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"signature":["up","dn"],"size":[245,200],"enriched":[6,2],"gsets":["REACTOME","REACTOME"],"bg":[3068,3068]},"columns":[{"id":".details","name":"","type":null,"headerClassName":"rctbl-outer-header","sortable":false,"resizable":false,"filterable":false,"searchable":false,"width":45,"align":"center","details":[{"name":"div","attribs":{"className":"rctbl-inner-obj"},"children":[{"name":"Reactable","attribs":{"data":{"label":["Amino acid transport across the plasma membrane","Tryptophan catabolism","Urea cycle","Recycling of bile acids and salts","Aspartate and asparagine metabolism","Phenylalanine metabolism"],"pval":[6.7e-20,1.6e-08,8.4e-06,2.1e-05,3.3e-05,0.00019],"fdr":[1.1e-16,1.2e-05,0.0044,0.0081,0.01,0.049],"signature":[245,245,245,245,245,245],"geneset":[33,14,10,16,12,6],"weighted_overlap":[22,9,6,7,6,4],"gene_hits":["SLC16A10;SLC1A4;SLC36A1;SLC36A2;SLC36A4;SLC38A1;SLC38A2;SLC38A4;SLC3A1;SLC3A2;SLC43A1;SLC43A2;SLC6A14;SLC6A15;SLC6A18;SLC6A19;SLC6A20;SLC7A1;SLC7A11;SLC7A2;SLC7A3;SLC7A5;SLC7A6;SLC7A7;SLC7A8;SLC7A9","AADAT;AFMID;HAAO;KMO;KYAT1;KYAT3;KYNU;SLC36A4;SLC3A2;SLC7A5","ARG1;ARG2;ASS1;NAGS;OTC;SLC25A15;SLC25A2","ABCB11;ABCC3;ALB;BAAT;SLC10A1;SLC10A2;SLCO1A2;SLCO1B1;SLCO1B3","ASNS;ASPA;ASPG;GOT1;GOT2;NAT8L;SLC25A12;SLC25A13","ASRGL1;KYAT1;PAH;PCBD1;QDPR"],"metabolite_hits":["Hydroxykynurenine;Ornithine;Phenylalanine;Proline;Aspartic acid;Cystine;Glycocholic acid;Kynurenine","2-Keto-glutaramic acid;Hydroxykynurenine;Kynurenic acid;Kynurenine;3-Hydroxyanthranilic acid;Ornithine;Phenylalanine;Proline;Cystine","Aspartic acid;N-Acetylmethionine;Ornithine","Aspartic acid;Phenylalanine;Proline;Glycochenodeoxycholic acid;Glycocholic acid;Glycoursodeoxycholic acid;Glycochenodeoxycholic acid 3-sulfate;Glycodeoxycholic acid 3-sulfate;Taurochenodeoxycholic acid 3-sulfate;Taurodeoxycholic acid 3-sulfate","Aspartic acid;Phenylalanine","2-Keto-glutaramic acid;Hydroxykynurenine;Kynurenic acid;Kynurenine;Aspartic acid;4a-Carbinolamine tetrahydrobiopterin;Phenylalanine"]},"columns":[{"id":"label","name":"Label","type":"character","headerClassName":"rctbl-inner-header","minWidth":300},{"id":"pval","name":"P-Value","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"fdr","name":"FDR","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"signature","name":"signature","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"geneset","name":"Geneset Size","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"weighted_overlap","name":"Weighted overlap","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"gene_hits","name":"Gene Hits","type":"character","headerClassName":"rctbl-inner-header"},{"id":"metabolite_hits","name":"metabolite_hits","type":"character","headerClassName":"rctbl-inner-header"}],"resizable":true,"searchable":true,"showPageSizeOptions":false,"compact":true,"className":"rctbl-inner-tbl","dataKey":"3e1994c35736a03741108dab0ccc22d7","static":false,"nested":true},"children":[]}]},{"name":"div","attribs":{"className":"rctbl-inner-obj"},"children":[{"name":"Reactable","attribs":{"data":{"label":["Pyrimidine catabolism","Transport of nucleosides and free purine and pyrimidine bases across the plasma membrane"],"pval":[1.2e-05,1.2e-05],"fdr":[0.0031,0.0031],"signature":[200,200],"geneset":[12,12],"weighted_overlap":[6,6],"gene_hits":["DPYD;NT5C;NT5C1A;NT5C3A;NT5E;NT5M;TYMP","SLC28A1;SLC28A2;SLC28A3;SLC29A1;SLC29A2;SLC29A3;SLC29A4"],"metabolite_hits":["Adenosine;D-Ribosylnicotinate;Thymine","Adenosine;Serotonin;Tryptamine;Tyramine;Thymine"]},"columns":[{"id":"label","name":"Label","type":"character","headerClassName":"rctbl-inner-header","minWidth":300},{"id":"pval","name":"P-Value","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"fdr","name":"FDR","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"signature","name":"signature","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"geneset","name":"Geneset Size","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"weighted_overlap","name":"Weighted overlap","type":"numeric","headerClassName":"rctbl-inner-header"},{"id":"gene_hits","name":"Gene Hits","type":"character","headerClassName":"rctbl-inner-header"},{"id":"metabolite_hits","name":"metabolite_hits","type":"character","headerClassName":"rctbl-inner-header"}],"resizable":true,"searchable":true,"showPageSizeOptions":false,"compact":true,"className":"rctbl-inner-tbl","dataKey":"a6aa1378e710389afd8053afce9209c0","static":false,"nested":true},"children":[]}]}]},{"id":"signature","name":"Signature","type":"character","headerClassName":"rctbl-outer-header","minWidth":300},{"id":"size","name":"Signature Size","type":"numeric","headerClassName":"rctbl-outer-header"},{"id":"enriched","name":"Enriched Genesets","type":"numeric","headerClassName":"rctbl-outer-header"},{"id":"gsets","name":"Genesets","type":"character","headerClassName":"rctbl-outer-header"},{"id":"bg","name":"Background","type":"numeric","headerClassName":"rctbl-outer-header"}],"resizable":true,"showPageSizeOptions":false,"onClick":"expand","nowrap":true,"className":"rctbl-outer-tbl","rowStyle":{"cursor":"pointer"},"dataKey":"c347758545cc70e53de289f50dd68f85"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div class="section level3">
<h3 id="sankey-diagram">Sankey diagram<a class="anchor" aria-label="anchor" href="#sankey-diagram"></a>
</h3>
<p>The <code><a href="../reference/sankey_plot.html">sankey_plot()</a></code> function generates a three-layer
Sankey diagrams connecting metabolites, the metabolite sets (MSets), and
the gene sets (GSets) derived from hypeR-GEM. Edge weights between the
metabolite and Msets layers reflect the number of shared metabolites,
while edges between the MSets and GSets layers reflect the overlap
between hypeR-GEM–mapped genes and genes in each GSets. The functions
takes the following key arguments:</p>
<ul>
<li><p><code>hypeR_GEM_obj</code>: The object returned by
<code><a href="../reference/signature2gene.html">signature2gene()</a></code> function.</p></li>
<li><p><code>msets_list</code>: A list of metabolite-sets. Each element
should be a named list where names are set names and values are
character vectors of metabolite identifiers matching the
<code>key</code> argument.</p></li>
<li><p><code>gsets_list</code>: A list of gene-sets. Each element should
be a named list where names are set names and values are character
vectors of metabolite identifiers matching the <code>key</code>
argument.</p></li>
<li><p><code>key</code>: Column name in
<code>hypeR_GEM_obj[["mapped_metabolite_signatures"]]</code> used as the
metabolite identifier.</p></li>
</ul>
<p>As an illustrative example, we applied this function to the
<code>hypeR-GEM_obj</code> constructed from age-associated metabolites
identified in the New England Centenarian Study (NECS) (See <a href="https://www.bumc.bu.edu/centenarian/" class="external-link">reference</a>). A curated
subset of enriched MSets from these metabolites, together with a curated
subset of enriched GSets derived from hypeR-GEM–mapped enzyme-coding
genes, were visualized in an integrative, multi-omics context.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"hypeR_object_NECS"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"metabolon_msets_list"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"gsets_list"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sankey_plot.html">sankey_plot</a></span><span class="op">(</span><span class="va">hypeR_object_NECS</span>, </span>
<span>                 msets_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">metabolon_msets_list</span><span class="op">$</span><span class="va">metabolonSupPthwy</span><span class="op">)</span>, </span>
<span>                 gsets_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">gsets_list</span><span class="op">$</span><span class="va">hallmark</span><span class="op">)</span>, </span>
<span>                 key <span class="op">=</span> <span class="st">"refmet_name"</span>,</span>
<span>                 font_size<span class="op">=</span><span class="fl">16</span>,</span>
<span>                 node_width<span class="op">=</span><span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="sankeyNetwork html-widget html-fill-item" id="htmlwidget-36aa3d2a04d42bbc2145" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-36aa3d2a04d42bbc2145">{"x":{"links":{"source":[0,1,1,0,1,0,1,1,2,2,2,2,2,2,2,3,4,4,4,4,4,4,5,5,5],"target":[2,2,3,4,4,5,5,6,7,8,9,10,11,12,13,12,7,8,9,10,12,11,8,11,9],"value":[5,40,5,19,25,1,2,2,10,14,6,4,10,5,7,1,21,18,11,16,5,2,3,1,1]},"nodes":{"name":["dn","up","Amino Acid","Carbohydrate","Lipid","Nucleotide","Xenobiotics","Fatty Acid Metabolism","Xenobiotic Metabolism","Bile Acid Metabolism","Peroxisome","Complement","Myogenesis","Coagulation"],"group":["dn","up","Amino Acid","Carbohydrate","Lipid","Nucleotide","Xenobiotics","Fatty Acid Metabolism","Xenobiotic Metabolism","Bile Acid Metabolism","Peroxisome","Complement","Myogenesis","Coagulation"]},"options":{"NodeID":"name","NodeGroup":"name","LinkGroup":null,"colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":16,"fontFamily":null,"nodeWidth":20,"nodePadding":10,"units":"","margin":{"top":null,"right":null,"bottom":null,"left":null},"iterations":32,"sinksRight":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level3">
<h3 id="export-to-excel-files">Export to Excel files<a class="anchor" aria-label="anchor" href="#export-to-excel-files"></a>
</h3>
<p>Mapping results can be exported to Excel files using the
<code><a href="../reference/hypGEM_to_excel.html">hypGEM_to_excel()</a></code> function:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hypGEM_to_excel.html">hypGEM_to_excel</a></span><span class="op">(</span><span class="va">hypeR_GEM_obj</span>, file <span class="op">=</span> <span class="st">"hypeR_GEM_non_directional.xlsx"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/hypGEM_to_excel.html">hypGEM_to_excel</a></span><span class="op">(</span><span class="va">hypeR_GEM_obj_di</span>, file <span class="op">=</span> <span class="st">"hypeR_GEM_directional.xlsx"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="section"></h3>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://github.com/Ziwei-Huang-BU" class="external-link">Ziwei Huang</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
